query/e2e_test.go

```diff
@@ -206,6 +206,31 @@ func TestE2EDuplicateTypes(t *testing.T) {
 			}}}
 
 	queryText := `EVENT SEQ(t0 e0, t0 e1) WHERE [attr]`
+
+	testEventSequence(t, queryText, events, true)
+}
+
+func TestE2ENegatedSequence(t *testing.T) {
+	events := []domain.Event{
+		&tEventImpl{
+			typ: "t0",
+			ts:  time.Date(2014, 1, 1, 0, 0, 20, 0, time.UTC),
+		},
+		&tEventImpl{
+			typ: "t0",
+			ts:  time.Date(2014, 1, 1, 0, 0, 20, 1, time.UTC),
+			attrs: map[string]interface{}{
+				"attr": "val",
+			},
+		},
+	}
+
+	queryText := `EVENT SEQ(t0 e0, !(t0 e1), t0 e2) WHERE e1.attr == "val"`
+
+	testEventSequence(t, queryText, events, false)
```

Comment:
You should be able to add this as another test case to [`TestE2E`](https://github.com/obeattie/sase/blob/master/query/e2e_test.go#L114).